### PR TITLE
[CARBONDATA-1326] Findbug fixes

### PR DESCRIPTION
**Pull request by ManoharVanam**
Findbug fixes

### CODE REVIEW

File: core/src/main/java/org/apache/carbondata/core/reader/CarbonDeleteFilesDataReader.java [hunk 1]
```diff
@@ -216,6 +198,17 @@ public DeleteDeltaBlockDetails getCompactedDeleteDeltaFileFromBlock(List<String>
     }
     return deleteDeltaResultSet;
   }
+  static class DeleteDeltaFileReaderCallable implements Callable<DeleteDeltaBlockDetails> {
```
Comment (xuchuanyin) [inline core/src/main/java/org/apache/carbondata/core/reader/CarbonDeleteFilesDataReader.java]: Why use `static` class ?

Comment (sraghunandan) [inline core/src/main/java/org/apache/carbondata/core/reader/CarbonDeleteFilesDataReader.java]: findbugs will complain as inner class with larger scope
can make it private

**Comment by CarbonDataQA** [comment]
SDV Build Success with Spark 2.1, Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/274/

**Comment by ravipesala** [comment]
LGTM

**Comment by jackylk** [comment]
LGTM

**Comment by sraghunandan** [comment]
LGTM

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/23/

**Comment by ManoharVanam** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/122/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/246/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/877/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1204/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/573/

**Pull request closed without merging**